flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -919,9 +920,33 @@ public final ExecutorService getAsyncOperationsThreadPool() {
 
 	@Override
 	public Future<Void> notifyCheckpointCompleteAsync(long checkpointId) {
-		return mailboxProcessor.getMailboxExecutor(TaskMailbox.MAX_PRIORITY).submit(
-				() -> notifyCheckpointComplete(checkpointId),
-				"checkpoint %d complete", checkpointId);
+		return notifyCheckpointOperation(
+			() -> notifyCheckpointComplete(checkpointId),
+			String.format("checkpoint %d complete", checkpointId));
+	}
+
+	@Override
+	public Future<Void> notifyCheckpointAbortAsync(long checkpointId) {
+		return notifyCheckpointOperation(
+			() -> subtaskCheckpointCoordinator.notifyCheckpointAborted(checkpointId, operatorChain, this::isRunning),
+			String.format("checkpoint %d aborted", checkpointId));
+	}
+
+	private Future<Void> notifyCheckpointOperation(RunnableWithException runnable, String description) {
```

Comment:
Could you re-use this method in `org.apache.flink.streaming.runtime.tasks.StreamTask#triggerCheckpointAsync`?

However I see there is a difference in the used mailbox executor (different priority), so if this would cause too much extra complexity I'm fine with leaving it as it is.

Reply:
`triggerCheckpointAsync` need to return `Future<Boolean>` while those operators just return `Future<Void>`. If we decide to change interfaces in `AbstractInvokable` to return both `Future<Boolean>`, I think this method could be re-used.

Reply:
Thanks for pointing it out. In that case I think there are a bit too many small differences and maybe it's better to keep it as it is.